### PR TITLE
Add cmake build for tflite label_image example.

### PR DESCRIPTION
**Pull request by JerryShih**
Add label_image build for cmake.

### CODE REVIEW

**Comment by JerryShih** [comment]
@mihaimaruseac
Could you please review this pr? The cmake build for label_image example.

**Comment by abattery** [comment]
@terryheo could you review this PR?

**Comment by terryheo** [comment]
@JerryShih Thanks for the contribution.
Can we still build benchmark tool under the tflite cmake build directory?

https://www.tensorflow.org/lite/guide/build_cmake#step_6_build_tensorflow_lite_benchmark_tool

File: tensorflow/lite/examples/label_image/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  ${TFLITE_SOURCE_DIR}/profiling/profile_summarizer.cc
+  ${TFLITE_SOURCE_DIR}/profiling/profile_summary_formatter.cc
+  ${TFLITE_SOURCE_DIR}/profiling/time.cc
+  ${TFLITE_SOURCE_DIR}/tools/command_line_flags.cc
+  ${TFLITE_SOURCE_DIR}/tools/delegates/default_execution_provider.cc
+  ${TFLITE_SOURCE_DIR}/tools/evaluation/utils.cc
+  ${TFLITE_SOURCE_DIR}/tools/tool_params.cc
+)
+
+if(TFLITE_ENABLE_XNNPACK)
```
Comment (terryheo) [inline tensorflow/lite/examples/label_image/CMakeLists.txt:35]: did you test these delegates?

File: tensorflow/lite/examples/label_image/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
```
Comment (terryheo) [inline tensorflow/lite/examples/label_image/CMakeLists.txt:2]: 2021?

**Comment by JerryShih** [comment]
> @JerryShih Thanks for the contribution.
> Can we still build benchmark tool under the tflite cmake build directory?
> 
> https://www.tensorflow.org/lite/guide/build_cmake#step_6_build_tensorflow_lite_benchmark_tool

Yes, we could still use the same target name to build that benchmark tool.

benchmark tool
```
cmd:
cmake --build . -j -t benchmark_model

log:
Scanning dependencies of target benchmark_model
...
[100%] Building CXX object tools/benchmark/CMakeFiles/benchmark_model.dir/__/__/profiling/profile_summarizer.cc.o
[100%] Linking CXX executable benchmark_model
[100%] Built target benchmark_model

binary path:
/tensorflow/tensorflow/lite/build$ ls tools/benchmark/
benchmark_model  CMakeFiles  cmake_install.cmake  Makefile
```

label_image example
```
cmd:
cmake --build . -j -t label_image

log:
Scanning dependencies of target label_image
...
[100%] Linking CXX executable label_image
[100%] Built target label_image

binary path:
/tensorflow/tensorflow/lite/build$ ls examples/label_image/
CMakeFiles  cmake_install.cmake  label_image  Makefile
```

File: tensorflow/lite/examples/label_image/CMakeLists.txt (hunk 2 above)
Comment (JerryShih) [reply tensorflow/lite/examples/label_image/CMakeLists.txt]: Fixed.

File: tensorflow/lite/g3doc/guide/build_cmake.md [hunk 3]
```diff
@@ -102,14 +102,18 @@ CMake project. Please refer the
 ["Create a CMake project which uses TensorFlow Lite"](#create_a_cmake_project_which_uses_tensorflow_lite)
 section.
 
-### Step 6. Build TensorFlow Lite Benchmark Tool
+### Step 6. Build TensorFlow Lite Benchmark Tool and Label Image Example
```
Comment (terryheo) [inline tensorflow/lite/g3doc/guide/build_cmake.md:105]: Let's rephrase "Label Image Example (Optional)"

File: tensorflow/lite/examples/label_image/CMakeLists.txt (hunk 1 above)
Comment (JerryShih) [reply tensorflow/lite/examples/label_image/CMakeLists.txt:35]: @terryheo
Yes, I add another commit(https://github.com/tensorflow/tensorflow/pull/47317/commits/79e6e84b602b7a42ddb8aaa589f16367a419b5c3) to show the delegation log.

I add the addition def "TFLITE_PROFILING_ENABLED" to show the profiling result in label_image.
```
diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index 74f1dc8..9dbfb0f 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -143,7 +143,7 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
   list(APPEND TF_TARGET_PRIVATE_OPTIONS "-Wno-deprecated-declarations")
 endif()
 # Additional compiler flags used when compiling TF Lite.
-set(TFLITE_TARGET_PUBLIC_OPTIONS "")
+set(TFLITE_TARGET_PUBLIC_OPTIONS "-DTFLITE_PROFILING_ENABLED")
 set(TFLITE_TARGET_PRIVATE_OPTIONS "")
 # Additional library dependencies based upon enabled features.
 set(TFLITE_TARGET_DEPENDENCIES "")	
```

The log:
```
/tensorflow/tensorflow/lite/build$ ./examples/label_image/label_image -m ~/mobilenet_v1_1.0_224.f32.tflite -l ~/labels.txt -i /tensorflow/tensorflow/lite/examples/label_image/testdata/grace_hopper.bmp -p 1 -x 1

INFO: Loaded model /home/jerrys/mobilenet_v1_1.0_224.f32.tflite
INFO: resolved reporter
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
INFO: Use XNNPACK acceleration.
INFO: Applied XNNPACK delegate.
ERROR: ModifyGraphWithDelegate is disallowed when graph is immutable.
ERROR: Ignoring failed application of the default TensorFlow Lite delegate indexed at 0.
INFO: invoked
INFO: average time: 6.421 ms
INFO:      6.411, Subgraph   0, Node   0, OpCode   3, CONV_2D
INFO:      6.388, Subgraph   0, Node  31, OpCode  51, DELEGATE
INFO:      0.002, Subgraph   0, Node  29, OpCode  43, SQUEEZE
INFO:      0.008, Subgraph   0, Node  32, OpCode  51, DELEGATE
INFO: 0.860174: 653 653:military uniform
INFO: 0.0481019: 907 907:Windsor tie
INFO: 0.00786707: 466 466:bulletproof vest
INFO: 0.00644937: 514 514:cornet, horn, trumpet, trump
INFO: 0.00608026: 543 543:drumstick
```

You could see the message:
```
INFO: Applied XNNPACK delegate.
ERROR: ModifyGraphWithDelegate is disallowed when graph is immutable.
ERROR: Ignoring failed application of the default TensorFlow Lite delegate indexed at 0.
```

The xnnpack backend was created successfully.

The "ModifyGraphWithDelegate is disallowed when graph is immutable" is another issue:https://github.com/tensorflow/tensorflow/issues/42757

Comment (JerryShih) [reply tensorflow/lite/examples/label_image/CMakeLists.txt:35]: The "DELEGATE" op in profiling result shows that the graph goes through delegation backend.

File: tensorflow/lite/g3doc/guide/build_cmake.md (hunk 3 above)
Comment (JerryShih) [reply tensorflow/lite/g3doc/guide/build_cmake.md]: Fixed.

File: tensorflow/lite/g3doc/guide/build_cmake.md [hunk 4]
```diff
@@ -102,7 +102,7 @@ CMake project. Please refer the
 ["Create a CMake project which uses TensorFlow Lite"](#create_a_cmake_project_which_uses_tensorflow_lite)
 section.
 
-### Step 6. Build TensorFlow Lite Benchmark Tool and Label Image Example
+### Step 6. Build TensorFlow Lite Benchmark Tool and Label Image Example(Optional)
```
Comment (terryheo) [inline tensorflow/lite/g3doc/guide/build_cmake.md:105]: Sorry, could you add one space before "(Optional)" ?

### Step 6. Build TensorFlow Lite Benchmark Tool and Label Image Example (Optional)

Comment (JerryShih) [reply tensorflow/lite/g3doc/guide/build_cmake.md:105]: fixed.

**Review by terryheo — APPROVED** [review]
LGTM

**Pull request merged**